### PR TITLE
Crosswords: clue URLs

### PR DESCRIPTION
**Pull request by OliverJAsh**
This gives clues a URL, e.g. `/crosswords/cryptic/26666#16-across`.

Clicking a clue from the list will navigate to the URL.

### CODE REVIEW

**Comment by crifmulholland** [comment]
love it!!!

**Comment by OliverJAsh** [comment]
One thought: the page has several uses of the hash fragment, i.e. clues and comment IDs. This don't conflict but in the future we might need a better protocol

File: static/src/javascripts/es6/projects/common/modules/crosswords/main.js [hunk 1]
```diff
@@ -347,14 +347,25 @@ class Crossword extends React.Component {
...
+        if (clues && clue) {
             this.focusHiddenInput(x, y);
 
             this.setState({
                 cellInFocus: { x: x, y: y },
                 directionOfEntry: direction
             });
+
+            // Side effect
+            window.location.hash = clue.id;
```
Comment (OliverJAsh) [inline static/src/javascripts/es6/projects/common/modules/crosswords/main.js]: The list of clues now link to the hash fragment, but we also need to update the URL when a cell is selected in the grid.

**Comment by gtrufitt** [comment]
:+1:  Nice!

**Pull request merged**